src/pages/take-quiz-page/take-quiz.tsx

```diff
@@ -1,7 +1,307 @@
+            };
+          }
+        });
+
+        QuizStore.editUserQuestionAnswer(question.id, {
+          ...question,
+          id: question.id,
+          answerList: [...arrayCopy],
+        });
+        return arrayCopy;
+      });
+    } else if (
+      questions[index]?.question?.questionType === QUESTION_TYPES.MULTIPLE.value
+    ) {
+      setQuestionAnswers((currentState) => {
+        let arrayCopy = [...currentState];
+        arrayCopy[indexAns].isCorrectAnswer =
+          !arrayCopy[indexAns].isCorrectAnswer;
+
+        QuizStore.editUserQuestionAnswer(question.id, {
+          ...question,
+          id: question.id,
+          answerList: [...arrayCopy],
+        });
+        return arrayCopy;
+      });
+    }
+  };
+
+  function changeElement(type: string) {
+    if (type === "next") {
+      setIndex(index + 1);
+      QuizStore.selectActiveQuestion(QuizStore.userAnswers[index + 1]?.id);
+    } else {
+      setIndex(index - 1);
+      QuizStore.selectActiveQuestion(QuizStore.userAnswers[index - 1]?.id);
+    }
+  }
+
+  const calculateFinalScore = () => {
+    let finalScore = 0;
+    questions.forEach((question, indexQ) => {
+      // Verify if the array of correct answers is equal with the user's answers
+      if (
+        JSON.stringify(question?.question?.answerList) ===
+        JSON.stringify(QuizStore.userAnswers[indexQ].answerList)
+      ) {
+        finalScore++;
+      }
+    });
+    QuizStore.finalScoreQuiz = finalScore;
+  };
+
+  useEffect(() => {
+    if (QuizStore.activeQuestion?.answerList) {
+      setQuestionAnswers(
+        QuizStore.activeQuestion?.answerList.map((answer) => {
+          return {
+            ...answer,
+            isCorrectAnswer: false,
+          };
+        })
+      );
+    }
+  }, [QuizStore.activeQuestionId, quizId]);
+
+  useEffect(() => {
+    // Check for any selected quiz
+    if (quizId) {
+      const docRef = doc(db, "quizzes", quizId);
+
+      getDoc(docRef).then((doc) => {
+        const docData = doc.data()?.quiz;
+        QuizStore.selectQuiz(quizId, "take");
+        setQuizName(docData.quizName);
+        setQuestions(docData.quizQuestions);
+
+        docData.quizQuestions.forEach((question: any) => {
+          const newAnswerList = question.question.answerList.map(
+            (answer: any) => {
+              return {
+                ...answer,
+                isCorrectAnswer: false,
+              };
+            }
+          );
+          QuizStore.editUserQuestionAnswer(question.id, {
+            ...question.question,
+            id: question.id,
+            answerList: [...newAnswerList],
+          });
+        });
+
+        QuizStore.selectActiveQuestion(docData.quizQuestions[0]?.id);
+        setLoading(false);
+      });
+    }
+
+    return () => {
+      QuizStore.deselectQuiz();
+    };
+  }, []);
+
+  function getElements(questions: any, index: number) {
+    return questions.slice(index, index + 1).map((question: any) => {
```

Comment:
From what it seems `questions.slice(index, index + 1)` returns the same as `questions[index]`. In this case the `map` is not necessary as it will always return just one element at a time. I think it would be simpler to change this function to take a single index as parameter and taking the question directly from `QuizStore.userAnswers`: `const question = QuizStore.userAnswers[index]`.